quickwit/quickwit-serve/src/test_utils/rest_client.rs

```diff
@@ -42,6 +42,25 @@ impl QuickwitRestClient {
         Self { root_url, client }
     }
 
+    pub async fn ingest_data(&self, index_id: &str, ndjson_doc: &str) -> anyhow::Result<()> {
+        let uri = format!("{}/api/v1/{index_id}/ingest", self.root_url)
+            .parse::<hyper::Uri>()
+            .unwrap();
+        let request = Request::builder()
+            .uri(uri)
+            .method("POST")
+            .header("content-type", "application/json")
+            .body(Body::from(ndjson_doc.to_string()))
+            .unwrap();
+        let response = self.client.request(request).await.unwrap();
+        if response.status() == StatusCode::OK {
+            return Ok(());
+        }
+        let body_bytes = hyper::body::to_bytes(response.into_body()).await.unwrap();
+        let body_string = String::from_utf8(body_bytes.to_vec()).unwrap();
+        Err(anyhow::anyhow!("error when creating index: {body_string}"))
```

Comment:
```suggestion
        let response = self.client.request(request).await.context("Failed to emit request")?;
        if response.status() != StatusCode::OK {
            let body_bytes = hyper::body::to_bytes(response.into_body()).await.unwrap();
            let body_str = std::str::from_utf8_lossy(&body_bytes);
            bail!("error when creating index: {body_str}"));
        }
        Ok(())
```

super minor espeically considered it is in tests.

early return on success is a bit twisted.